include/oneapi/dpl/internal/async_impl/glue_async_impl.h

```diff
@@ -16,18 +16,30 @@
 #ifndef _ONEDPL_GLUE_ASYNC_IMPL_H
 #define _ONEDPL_GLUE_ASYNC_IMPL_H
 
-#include "../async_extension_defs.h"
+#if _ONEDPL_BACKEND_SYCL
+#    include "../async_extension_defs.h"
 
-#if _ONEDPL_HETERO_BACKEND
-#    include "async_impl_hetero.h"
-#endif
+#    if _ONEDPL_HETERO_BACKEND
+#        include "async_impl_hetero.h"
+#    endif
+#endif // _ONEDPL_BACKEND_SYCL
 
 namespace oneapi
 {
 namespace dpl
 {
 namespace experimental
 {
+#if _ONEDPL_BACKEND_SYCL
+
+// [wait_for_all]
+template <typename... _Ts>
+oneapi::dpl::__internal::__enable_if_convertible_to_events<void, _Ts...>
+wait_for_all(_Ts&&... __events)
```

Comment:
The function has sycl specific code.... I would suggest moving it to `async_impl_hetero.h` f.e.

Reply:
`wait_for_all` is a public API, and so it equally makes sense to have it in this header.

Perhaps, designing and proposing some "backend API" for waiting makes sense, but for this patch I would just add a TODO comment or a task to address this later.

Reply:
```TODO``` comment inserted into wait_for_all(_Ts&&... __events) implementation